data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,617 @@
+
+# sqrt 9.0;;
+- : float = 3.
+
+# sqrt @@ 9.0;;
+- : float = 3.
+```
+
+The application operator applies a parameter (on the right) to a function (on the left). It is useful when chaining several calls, as it avoids writing parentheses. Here is what it can look like:
+```ocaml
+# truncate (sqrt (float_of_int (int_of_string "81")));;
+- : int = 9
+
+# truncate @@ sqrt @@ float_of_int @@ int_of_string "81";;
+- : int = 9
+```
+
+The pipe operator also allows avoiding parentheses but in reversed order:
+```ocaml
+# ( |> );;
+- : 'a -> ('a -> 'b) -> 'b = <fun>
+
+# "81" |> int_of_string |> float_of_int |> sqrt |> truncate;;
+- : int = 9
+```
+This is just like a Unix shell pipe.
+
+## Anonymous Functions
+
+As citizens of the same level as other values, functions don't have to be bound to a name to exist (although some must, but this will be explained later). Function values not bound to names are called _anonymous functions_. Here are a couple of examples:
+```ocaml
+# fun x -> x;;
+- : 'a -> 'a = <fun>
+
+# fun x -> x * x;;
+- : int -> int = <fun>
+
+# fun s t -> s ^ " " ^ t ;;
+- : string -> string-> string = <fun>
+
+# function [] -> None | x :: _ -> Some x;;
+- : 'a list -> 'a option = <fun>
+```
+
+In order, here is what they do:
+- The identity function takes anything, returns it unchanged.
+- The square function takes an integer, returns its square.
+- The function taking two strings returns their concatenation with a space character in between.
+- A function taking a list returns `None`, if the list is empty, and the tail of the list otherwise.
+
+## Defining Global Functions
+
+Defining a global function isn't very different from creating a definition and binding a value to a name. The only difference is that no computation takes place. The expression is turned into a function value and bound to the name.
+```ocaml
+# let f = fun x -> x * x;;
+val f : int -> int = <fun>
+
+# let f x = x * x;;
+val f : int -> int = <fun>
+```
+
+These two definitions are the same. The former explicitly binds the anonymous function to a name. The latter is the same, but it uses a more compact syntax and avoids the `fun` keyword and the arrow symbol.
+
+## Defining Local Functions
+
+A function may be defined locally. Just like any local definition, meaning the function is only available inside its attached expression. Although the local functions are often defined inside the function's scope, this is not a requirement. Here is a local function inside a toplevel expression:
+```ocaml
+# let sq x = x * x in sq 7 * sq 7;;
+- : int = 2401
+
+# sq;;
+Error: Unbound value sq
+```
+
+## Closures
+
+*Closures* are directly related to shadowing. As introduced in [Same-Level Shadowing](#same-level-shadowing] section, a closure uses the preceding name binding, even after that name is redefined. In other words, it's *enclosed* between two different definitions; it will never compute the name binding that comes after it. That's why it is called a closure.
```

Comment:
```suggestion
As introduced in [Same-Level Shadowing](#same-level-shadowing] section, a *closure* uses the preceding name binding, even after that name is redefined. In other words, it's *enclosed* between two different definitions; it will never compute the name binding that comes after it. That's why it is called a closure.
```